### PR TITLE
[flang] Move `genCommonBlockMember` from OpenMP to ConvertVariable, NFC

### PR DESCRIPTION
**Pull request by kparzysz**
The function `genCommonBlockMember` is not specific to OpenMP, and it could very well be a common utility. Move it to ConvertVariable.cpp where it logically belongs.

### CODE REVIEW

**Comment by clementval** [comment]
If there is no use of it elsewhere it can stay in OpenMP.cpp

File: flang/lib/Lower/OpenMP.cpp [hunk 1]
```diff
@@ -1959,31 +1959,6 @@ static mlir::Operation *getCompareFromReductionOp(mlir::Operation *reductionOp,
   return nullptr;
 }
 
-/// The COMMON block is a global structure. \p commonValue is the base address
-/// of the COMMON block. As the offset from the symbol \p sym, generate the
-/// COMMON block member value (commonValue + offset) for the symbol.
-/// FIXME: Share the code with `instantiateCommon` in ConvertVariable.cpp.
```
Comment (kiranchandramohan) [inline flang/lib/Lower/OpenMP.cpp:1965]: If you can address the FIXME that would be a good reason to move the code.

**Comment by kparzysz** [comment]
The build failures are caused by lit on Windows not being able to open four of the test files, e.g.
```
OSError: [Errno 22] Invalid argument: 'C:\\ws\\src\\flang\\test\\Lower\\OpenMP\\FIR\\declare-target-implicit-func-and-subr-cap-enter.f90'
```

**Review by kiranchandramohan — APPROVED** [review]
Thanks @kparzysz for this cleanup. LGTM. Please wait for either @clementval or @jeanPerier.

**Comment by kiranchandramohan** [comment]
> The build failures are caused by lit on Windows not being able to open four of the test files, e.g.
> 
> ```
> OSError: [Errno 22] Invalid argument: 'C:\\ws\\src\\flang\\test\\Lower\\OpenMP\\FIR\\declare-target-implicit-func-and-subr-cap-enter.f90'
> ```

Would a rebase solve the issue?

FYI @shraiysh.

**Review by jeanPerier — APPROVED** [review]
LGTM

**Pull request merged**